### PR TITLE
Feature/workflows

### PR DESCRIPTION
**Pull request by MarkArts**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/workflows/lint.yml') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - run: npm install
+
+      - name: Run prettier
+        run: npm run lint
```
Comment (macostea) [inline .github/workflows/lint.yml]: Wouldn't it be a better idea to use the prettier action? The npm caching here is also based on the lint.yml file which does not have the proper declaration for prettier so you'll always have the same npm cache

File: .github/workflows/on_pr.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+on:
+  workflow_call:
+
+jobs:
+  prettier_deploy_config:
+    runs-on: [self-hosted, ubuntu20.04-self]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actionsx/prettier@v2
```
Comment (macostea) [inline .github/workflows/on_pr.yml:11]: What's the use of this and lint.yml too?

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "yaml.schemas": {
+    "https://json.schemastore.org/github-workflow.json": [
+      "file:///home/mark/projects/github-actions/workflows/build_and_deploy.yml",
+      "file:///home/mark/projects/github-actions/workflows/push.yml",
+      "file:///home/mark/projects/github-actions/workflows/lint_and_diff.yml"
```
Comment (macostea) [inline .vscode/settings.json]: Does this file exist?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (MarkArts) [reply .github/workflows/lint.yml]: good catch 👍

File: .vscode/settings.json (hunk 3 above)
Comment (MarkArts) [reply .vscode/settings.json]: update files

File: .github/workflows/on_pr.yml (hunk 2 above)
Comment (MarkArts) [reply .github/workflows/on_pr.yml:11]: this is the workflow that will be include by other repos (and won't be used by this repo). 
it lints the ./deploy config for all repos. 

the lint.yml will lint this project. 

on_pr and on_push have to be in .github/workflows btw so no option to split the workflows

**Review by macostea — APPROVED** [review]

**Pull request merged**